test/unit/deadline_job_attachments/api/test_manifest_merge.py

```diff
@@ -0,0 +1,61 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import os
+import json
+
+from typing import Optional
+
+from deadline.job_attachments.api.manifest import _manifest_merge
+from deadline.job_attachments.models import ManifestMerge
+
+
+class TestMergeAPI:
+
+    def test_merge_same_file(self, temp_dir, test_manifest_one):
+        """
+        Merge with one manifest file
+        """
+        # Given
+        merge_dir = os.path.join(temp_dir, "merge")
+        manifest_one = os.path.join(temp_dir, "manifest_1")
+
+        with open(manifest_one, "w", encoding="utf8") as f:
+            json.dump(test_manifest_one, f)
+
+        # When
+        manifest_merge: Optional[ManifestMerge] = _manifest_merge(
+            root=temp_dir, manifest_files=[manifest_one], destination=merge_dir, name="merge"
+        )
+
+        # Then
+        assert manifest_merge is not None
+        assert manifest_merge.manifest_root == temp_dir
+        assert merge_dir in manifest_merge.local_manifest_path
+
+    def test_diff_new_files(self, temp_dir, test_manifest_one, test_manifest_two):
```

Comment:
```suggestion
    def test_merge_different_files(self, temp_dir, test_manifest_one, test_manifest_two):
```

nit: name is a bit confusing/misleading. Sounds like we are testing a `diff` functionality